icechunk/src/repository.rs

```diff
@@ -1112,6 +1114,17 @@ async fn all_chunks<'a>(
     Ok(existing_array_chunks.chain(new_array_chunks))
 }
 
+pub async fn raise_if_invalid_snapshot_id(
```

Comment:
nitpicking: I think Storage is enough here? No need for Send + Sync?

Reply:
apparently required:
```
error[E0277]: `dyn icechunk::Storage` cannot be shared between threads safely
    --> icechunk-python/src/lib.rs:364:9
     |
364  | /         pyo3_asyncio_0_21::tokio::future_into_py(py, async move {
365  | |             do_checkout_snapshot(store, snapshot_id).await
366  | |         })
     | |__________^ `dyn icechunk::Storage` cannot be shared between threads safely
     |
     = help: the trait `std::marker::Sync` is not implemented for `dyn icechunk::Storage`, which is required by `{async block@icechunk-python/src/lib.rs:364:54: 364:64}: std::marker::Send`
     = note: required for `&dyn icechunk::Storage` to implement `std::marker::Send`
note: required because it's used within this `async` fn body
```

Reply:
But, I don't get it. It errors in an unrelated function we are not calling?